packages/backend/src/apps/m365-excel/FOR_RELEASE_PERIOD_ONLY.ts

```diff
@@ -0,0 +1,55 @@
+import type { IGlobalVariable } from '@plumber/types'
+
+import { RateLimiterRedis, RateLimiterRes } from 'rate-limiter-flexible'
+
+import { M365_EXCEL_INTERVAL_BETWEEN_ACTIONS_MS } from '@/config/app-env-vars/m365'
+import { createRedisClient, REDIS_DB_INDEX } from '@/config/redis'
+import StepError from '@/errors/step'
+
+const redisClient = createRedisClient(REDIS_DB_INDEX.RATE_LIMIT)
+
+// Rate limiter to control potential thundering herd during release priod
+// --
+// This allows a burst from test steps up to 2x our allocated M365 quota.
+//
+// Instead of rate limiting per second, we allow spikes in 15 second windows to
+// make the UX not _as_ bad.
+//
+// (e.g. thundering herd spams tests within the 1st 10 seconds and use up all
+// quota allocated for 15s, but they don't complain since they're busy
+// publishing or thinking about why they had an error)
+const ACTIONS_PER_SECOND = 1000 / M365_EXCEL_INTERVAL_BETWEEN_ACTIONS_MS
+const DURATION_SECONDS = 30
+const ACTIONS_PER_DURATION = Math.floor(ACTIONS_PER_SECOND * DURATION_SECONDS)
+
+const rateLimiter = new RateLimiterRedis({
+  points: ACTIONS_PER_DURATION,
+  duration: DURATION_SECONDS,
+  keyPrefix: 'm365-release-limiter',
+  storeClient: redisClient,
+})
+
+export async function RATE_LIMIT_FOR_RELEASE_ONLY_REMOVE_AFTER_JULY_2024(
+  $?: IGlobalVariable,
+) {
+  try {
+    await rateLimiter.consume(1)
```

Comment:
question: it doesn't matter what key you pass in here?

Reply:
yeaa a bit confusing, i thought that was the points to consume. @ogp-weeloong can add in a comment that this key is arbitrary and that the rate limit is global

Reply:
actually no, I screwed up. The 1st arg is the key, but i mistakenly thought it was number of points to consume.

it's only because this is intended to be a global limit, thats why testing didn't catch this. I wish `consume` typed keys as strings =.=. I'm going to update the code to a fixed constant key.